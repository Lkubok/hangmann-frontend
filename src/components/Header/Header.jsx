import React, { Component } from "react";
import { connect } from "react-redux";
import "./Header.scss";

export class Header extends Component {
  render() {
    return (
      <header className="app-header">
        <img src="./img/logo.png" alt="" />
        {this.props.isLogged && (
          <h5 className="user">
            Logged as: <span>{this.props.userName}</span>
          </h5>
        )}
      </header>
    );
  }
}

const mapStateToProps = state => ({
  userName: state.appParamsReducer.userName,
  isLogged: state.appParamsReducer.isLogged
});

export default connect(
  mapStateToProps,
  null
)(Header);
