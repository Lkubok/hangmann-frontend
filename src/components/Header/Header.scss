@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";

.app-header {
  @include basic-app-padding;
  @include flex-cl-row;
  @include basic-app-margin;
  background-color: $back-color;
  color: $primary-color;
  padding-top: 10px;
  padding-bottom: 10px;
  justify-content: space-between;
}
.user {
  color: $grey-color;
  font-size: 14px;
  margin-top: 20px;
  span {
    text-transform: uppercase;
    color: $primary-color;
    margin: 10px 20px;
  }
}
