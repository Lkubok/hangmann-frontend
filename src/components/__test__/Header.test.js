import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Header } from "../Header/Header";

Enzyme.configure({ adapter: new Adapter() });

describe("Header component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
