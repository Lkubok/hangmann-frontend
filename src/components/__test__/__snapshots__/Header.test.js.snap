// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should match snapshot 1`] = `
<header
  className="app-header"
>
  <img
    alt=""
    src="./img/logo.png"
  />
</header>
`;
